tests/v2/validation/workloads/workload_test.go

```diff
@@ -192,6 +200,159 @@ func (w *WorkloadTestSuite) TestWorkloadStatefulset() {
+	require.NoError(w.T(), err)
+
+	revisions, err := deployment.GetRolloutHistoryDeployment(w.client, w.cluster.ID, namespace.Name, nginxDeployment)
+	require.NoError(w.T(), err)
+	require.Equal(w.T(), []string{"1"}, revisions)
+
+	countPods, err := pods.CountPodContainerRunningByImage(w.client, w.cluster.ID, namespace.Name, "nginx")
+	require.NoError(w.T(), err)
+	require.Equal(w.T(), 2, countPods)
+
+	containerName := namegen.AppendRandomString("updatetestcontainer")
+	newContainerTemplate := workloads.NewContainer(containerName,
+		"redis",
+		corev1.PullAlways,
+		[]corev1.VolumeMount{},
+		[]corev1.EnvFromSource{},
+		nil,
+		nil,
+		nil,
+	)
+	redisDeployment := nginxDeployment.DeepCopy()
+	redisDeployment.Spec.Template.Spec.Containers = []corev1.Container{newContainerTemplate}
+
+	redisDeployment, err = deployment.UpdateDeployment(w.client, w.cluster.ID, namespace.Name, redisDeployment)
+	require.NoError(w.T(), err)
+
+	err = charts.WatchAndWaitDeployments(w.client, w.cluster.ID, namespace.Name, metav1.ListOptions{
+		FieldSelector: "metadata.name=" + redisDeployment.Name,
+	})
+	require.NoError(w.T(), err)
+
+	err = pods.WatchAndWaitPodContainerRunning(w.client, w.cluster.ID, namespace.Name, redisDeployment)
+	require.NoError(w.T(), err)
+
+	revisions, err = deployment.GetRolloutHistoryDeployment(w.client, w.cluster.ID, namespace.Name, redisDeployment)
+	require.NoError(w.T(), err)
+	require.Equal(w.T(), []string{"1", "2"}, revisions)
+
+	countPods, err = pods.CountPodContainerRunningByImage(w.client, w.cluster.ID, namespace.Name, "redis")
+	require.NoError(w.T(), err)
+	require.Equal(w.T(), 2, countPods)
+
+	containerName = namegen.AppendRandomString("updatetestcontainertwo")
+	newContainerTemplate = workloads.NewContainer(containerName,
+		"ubuntu",
+		corev1.PullAlways,
+		[]corev1.VolumeMount{},
+		[]corev1.EnvFromSource{},
+		nil,
+		nil,
+		nil,
+	)
+	newContainerTemplate.TTY = true
+	newContainerTemplate.Stdin = true
+	ubuntuDeployment := redisDeployment.DeepCopy()
+	ubuntuDeployment.Spec.Template.Spec.Containers = []corev1.Container{newContainerTemplate}
+
+	_, err = deployment.UpdateDeployment(w.client, w.cluster.ID, namespace.Name, ubuntuDeployment)
+	require.NoError(w.T(), err)
+
+	err = charts.WatchAndWaitDeployments(w.client, w.cluster.ID, namespace.Name, metav1.ListOptions{
+		FieldSelector: "metadata.name=" + ubuntuDeployment.Name,
+	})
+	require.NoError(w.T(), err)
+
+	err = pods.WatchAndWaitPodContainerRunning(w.client, w.cluster.ID, namespace.Name, ubuntuDeployment)
+	require.NoError(w.T(), err)
+
+	revisions, err = deployment.GetRolloutHistoryDeployment(w.client, w.cluster.ID, namespace.Name, ubuntuDeployment)
+	require.NoError(w.T(), err)
+	require.Equal(w.T(), []string{"1", "2", "3"}, revisions)
+
+	countPods, err = pods.CountPodContainerRunningByImage(w.client, w.cluster.ID, namespace.Name, "ubuntu")
+	require.NoError(w.T(), err)
+	require.Equal(w.T(), 2, countPods)
```

Comment:
this is repeated quite a bit in here. Would it be worth writing a short helper function since these 2 are called in tandem so often? something like 'ValidateRolloutHistory(client, expectedRevisions, expectedContainerCount) error {...}`

Reply:
It's a good ideia, I will create a private function in the workloads_test